Explorer/Assets/DCL/CharacterPreview/ICharacterPreviewFactory.cs

```diff
@@ -0,0 +1,23 @@
+using Arch.Core;
+using DCL.Optimization.Pools;
+using UnityEngine;
+
+namespace DCL.CharacterPreview
+{    /// <summary>--
+    ///     Check ICharacterPreviewFactory in the old renderer
+    /// </summary>
+    public interface ICharacterPreviewFactory
+    {
+        CharacterPreviewController Create(World world, IComponentPoolsRegistry poolsRegistry, RenderTexture targetTexture,  CharacterPreviewInputEventBus inputEventBus);
+    }
+
+    public class CharacterPreviewFactory : ICharacterPreviewFactory
+    {
+        public CharacterPreviewController Create(World world, IComponentPoolsRegistry poolsRegistry, RenderTexture targetTexture, CharacterPreviewInputEventBus inputEventBus)
+        {
+            var container = (CharacterPreviewContainer)poolsRegistry.GetPool(typeof(CharacterPreviewContainer)).Rent();
```

Comment:
Resolve `IComponentPool<CharacterPreviewContainer>` once in the `ctor`. You don't need to pass the full registry every time, moreover, it's configured only once: you will never receive a different instance after resolution.
There is a generic version `IComponentPool<T> GetReferenceTypePool<T>() where T: class;`

Reply:
Fixed!